### ISSUE TITLE
Returning scalars from NDData

### ISSUE THREAD

**Issue by embray** [astropy/astropy#1314]
Currently if one wants to return a scalar value from an NDData object it's a bit of a mess:

```
In [1]: from astropy.nddata import NDData

In [2]: d = NDData([1, 2, 3])

In [3]: d[0]
Out[3]: <astropy.nddata.nddata.NDData at 0x2b33e90>

In [4]: d[0].data
Out[4]: array(1)

In [5]: d[0].data.item()
Out[5]: 1
```

Right now even a scalar is wrapped in the `NDData` class with metadata and everything.  I realize there are issues with this involving masks, error propagation, units (for arrays with units a scalar can just be returned as a `Quantity` object).  But I think that right now for simple cases like, "I just want an array with metadata attached" the current behavior is bad.  It would be better to implement reasonable behavior for now than to leave the class in this somewhat unusable state in anticipation of "eventually" solving all conceivable problems with how the class should ideally work.

**Comment by wkerzendorf** [comment]
@embray still want to leave it open?

**Comment by embray** [comment]
Hmm, well, I don't know if the NDData APE mentions this (probably not), but it might be nice if we required that when a slice returns a scalar it returns an actual scalar instead of a zero-dimensional array.

**Comment by mwcraig** [comment]
Closing, since it appears to be fixed in the latest (v1.3) nddata:

```python
In [2]: from astropy.nddata import NDData

In [3]: d = NDData([1, 2, 3])

In [4]: d[0]
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-17371c6688f6> in <module>()
----> 1 d[0]

TypeError: 'NDData' object does not support indexing

In [5]: d.data[0]
Out[5]: 1
```
